cheroot/server.py

```diff
@@ -2056,14 +2068,22 @@ def interrupt(self):
         """Flag interrupt of the server."""
         return self._interrupt
 
+    @property
+    def _stopping_for_interrupt(self):
+        """Return whether the server is responding to an interrupt."""
+        return self._interrupt is _STOPPING_FOR_INTERRUPT
+
     @interrupt.setter
     def interrupt(self, interrupt):
-        """Perform the shutdown of this server and save the exception."""
-        self._interrupt = True
+        """Perform the shutdown of this server and save the exception.
+
+        Typically invoked by a worker thread in threadpool.py,
+        the exception is raised from the thread running serve()
+        once self.stop() has completed.
```

Comment:
Let's reference the actual method docs in sphinx:
```suggestion
        the exception is raised from the thread running :py:meth:`serve`
        once :py:meth:`stop` has completed.
```